DatadogInternal/Sources/NetworkInstrumentation/TracePropagationHeadersWriter.swift

```diff
@@ -25,7 +25,18 @@ public enum TraceSamplingStrategy {
         case .headBased:
             return DeterministicSampler(shouldSample: traceContext.isKept, samplingRate: traceContext.sampleRate)
         case .custom(let sampleRate):
-            return DeterministicSampler(baseId: traceContext.traceID.idLo, samplingRate: sampleRate)
+            let sessionBasedId = if let sessionId = traceContext.rumSessionId {
+                // for a UUID with value aaaaaaaa-bbbb-Mccc-Nddd-1234567890ab
+                // we use as the base id the last part : 0x1234567890ab
+                if let lastSegment = sessionId.split(separator: "-").last {
+                    UInt64(lastSegment, radix: 16)
+                } else {
+                    Optional<UInt64>.none
+                }
+            } else {
+                Optional<UInt64>.none
+            }
+            return DeterministicSampler(baseId: sessionBasedId ?? traceContext.traceID.idLo, samplingRate: sampleRate)
```

Comment:
nit/ Here a more idiomatic way:
```suggestion
            // for a UUID with value aaaaaaaa-bbbb-Mccc-Nddd-1234567890ab
            // we use as the base id the last part : 0x1234567890ab
            let baseId = traceContext.rumSessionId
                .flatMap { $0.split(separator: "-").last }
                .flatMap { UInt64($0, radix: 16) }
            ?? traceContext.traceID.idLo

            return DeterministicSampler(
                baseId: baseId,
                samplingRate: sampleRate
            )
```

Reply:
Thanks, done :)